Added linked locale messages feature. (#280)
aissat/easy_localization: README.md

var money = plural('money', 10.23) // output: You have 10.23 dollars
```

### Linked translations:

If there's a translation key that will always have the same concrete text as another one you can just link to it. To link to another translation key, all you have to do is to prefix its contents with an `@:` sign followed by the full name of the translation key including the namespace you want to link to.

Example:
```json
{
  ...
  "example": {
    "hello": "Hello",
    "world": "World!",
    "helloWorld": "@:example.hello @:example.world"
  }
  ...
}
```

```dart
print('example.helloWorld'.tr()); //Output: Hello World!
```

You can also do nested anonymous and named arguments inside the linked messages.

Example:

```json
{
  ...
  "date": "{currentDate}.",
  "dateLogging": "INFO: the date today is @:date"
  ...
}
```
```dart
print('dateLogging'.tr(namedArguments: {'currentDate': DateTime.now().toIso8601String()})); //Output: INFO: the date today is 2020-11-27T16:40:42.657.
```

### Formatting linked translations:

Formatting linked locale messages

> If the language distinguish cases of character, you may need control the case of the linked locale messages. Linked messages can be formatted with modifier @.modifier:key

Reviewer: ```suggestion
If the language distinguishes cases of character, you may need to control the case of the linked locale messages. Linked messages can be formatted with modifier `@.modifier:key`
```

The below modifiers are available currently.

> - upper: Uppercase all characters in the linked message.

Reviewer: ```suggestion
- `upper`: Uppercase all characters in the linked message.
```

> - lower: Lowercase all characters in the linked message.

Reviewer: ```suggestion
- `lower`: Lowercase all characters in the linked message.
```

> - capitalize: Capitalize the first character in the linked message.

Reviewer: ```suggestion
- `capitalize`: Capitalize the first character in the linked message.
```
